docs/design/jet/023-kafka-connect-source-connectors.md

```diff
@@ -0,0 +1,339 @@
+        pipeline.readFrom(KafkaConnectSources.connect(randomProperties))
+                .withoutTimestamps()
+                .map(record -> Values.convertToString(record.valueSchema(), record.value()))
+                .writeTo(AssertionSinks.assertCollectedEventually(60,
+                        list -> assertEquals(ITEM_COUNT, list.size())));
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.addJar(Objects.requireNonNull(this.getClass()
+                .getClassLoader()
+                .getResource("random-connector-1.0-SNAPSHOT.jar"))
+        .getPath()
+        );
+
+        Job job = createHazelcastInstance().getJet().newJob(pipeline, jobConfig);
+```
+
+#### Client Related Changes
+
+No changes required
+
+### Technical Design
+
+In the `hazelcast-jet-contrib` repository we already have an initial implementation of the `KafkaConnectSources`.
+
+`KafkaConnectSources` instantiates the `KafkaConnectSource`, which is used then to create as task object which performs
+the actual ingestion logic.
+Everything is orchestrated by the `SourceBuilder`
+methods: `createFn`, `fillBufferFn`, `createSnapshotFn`, `restoreSnapshotFn`, `destroyFn`.
+
+The current implementation uses `SourceBuilder` to create a stream source of `SourceRecord`.
+`SourceRecord` is Kafka data type returned by the Kafka Source Connectors for further processing.
+
+The plan is to move the code to the main hazelcast repository and make it production ready with the following steps:
+
+- Move Kafka Connect to Hazelcast Extensions module
+- Add Kafka connect source scaling (currently Hazelcast Platform will instantiate a single task for the specified source
+  in the cluster)
+- Add unit tests for Kafka connect source
+- Add integration tests for Kafka connect source
+- Add documentation to official hazelcast documentation
+
+There are also nice to have features:
+
+- Expose Kafka connect as a SQL mapping
+- Use custom processor classpath if needed
+- Add Kafka connect metrics
+- Add benchmarks for Kafka connect source
+- Add Soak tests for Kafka connect source
+
+#### Fault-Tolerance
+
+The Kafka Connect connectors driven by Jet are participating to store their state snapshots (e.g partition offsets +
+any metadata which they might have to recover/restart) in Jet. This way when the job is restarted they can recover
+their state and continue to consume from where they left off. Since implementations may vary between Kafka Connect
+modules, each will have different behaviors when there is a failure. Please refer to the documentation of Kafka Connect
+connector of your choice for detailed information.
+
+#### Questions about the change
+
+- What components in Hazelcast need to change? How do they change?
+  * N/A
+
+- How does this work in an on-prem deployment?
+  * N/A
+
+- How about on AWS and Kubernetes, platform operator?
+  * As long as necessary jars are on the classpath it doesn't have deployment requirements.
```

Comment:
For  new HZ instance, dont we need to include any of the kafka connect components in the YAML file ?
For a running / existing HZ member, can you add Kafka connect component - dynamically without having to restart ?
How do you deploy these configurations from one environment to other ?